sansarip/owlbear
# #42 Support multi-language parsing

See #42.

**Optional Test steps**

These test steps are optional and really only for educational purposes, because the updated `test-runner` should load in multiple wasms. If the tests pass then that means the changes were successful for this issue.

1. Start the project (aka jack in)
2. Once CLJS compilation is finished, boot up the node environment with `npm run cljs-repl`
3. In your CLJS REPL, hop into a namespace of your choosing, e.g. [slurp](https://github.com/sansarip/owlbear/blob/d28867087dddf00443a8697aafb81430c3690229/src/cljs/owlbear/html/edit/slurp.cljs)
4. Load/register a language wasm of your choosing [other than the base Tree-sitter wasm]. The wasms are located in [resources](https://github.com/sansarip/owlbear/tree/d28867087dddf00443a8697aafb81430c3690229/resources). e.g. `(obp/load-language-wasm! :typescript "resources/tree-sitter-typescript.wasm")` The first argument can be anything, just remember it!
5. Parse some code! e.g. `(obp/src->tree "const foo = 'hello world';" :typescript)` The second arg has to be a language name that you've previously loaded/registered with a `load-language-wasm!` call. This should return a `Tree` object!
6. You should be able to load and parse more languages in your REPL too!
